### ISSUE TITLE
Fix all tests in `test/unit/test-ie-media-bug.js` that rely on a global `sinon` sandbox

### ISSUE THREAD

**Issue by rsimha** [ampproject/amphtml#25590]
In #23842 and #25587, we're eliminating the use of the global `sinon` sandbox in our tests. The new way to access the sandbox is via `window.sandbox` for legacy `describe` tests, and via `env.sandbox` for `describes` tests. (There's a new presubmit check to enforce this.)

When the cleanup was done for `test/unit/test-ie-media-bug.js` tests, a couple of tests are no longer passing, likely because they depended on the presence of a global `sinon` sandbox in some indirect way. These tests will be skipped by #25587.

This issue is to make sure the tests are fixed and unskipped.
 
/cc @ampproject/wg-ui-and-a11y

**Comment by rsimha** [comment]
**Failure logs:**

```
rsimha@bobdylan (2019-11-14-GlobalSinon):~/src/amphtml$ gulp unit --headless --files test/unit/test-ie-media-bug.js
[14:48:43] Using gulpfile ~/src/amphtml/gulpfile.js
[14:48:43] Starting 'unit'...
[14:48:43] Run gulp help to see a list of all test flags.
[14:48:43] ⤷ Use --nohelp to silence these messages.
[14:48:43] ⤷ Use --local_changes to run unit tests from files commited to the local branch.
[14:48:43] Running tests against minified code.
[14:48:43] --headless: Running tests in a headless Chrome window.
[14:48:43] --files: Running tests in the file(s): test/unit/test-ie-media-bug.js
[14:48:44] All packages in node_modules are up to date.
[14:48:46] Recompiled all CSS files into build/ (2.397 s)
[14:48:47] Compiled Jison parsers into build/parsers/ (662 ms)
[14:48:47] Started AMP Test Server at http://localhost:8081
[14:48:47] Running gulp unit. Press Ctrl + C to cancel...
Deprecated private createPreprocessor() API
[14:48:47] Running tests locally...

START:
[14:48:49] Transforming tests with browserify...
...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

[14:49:13] Done. Running tests...


[14:49:14] HeadlessChrome 78.0.3904 (Linux 0.0.0): STARTED
  ie-media-bug
    ● should bypass polling for non-IE browsers
    ● should bypass polling when matchMedia is not broken
    ● should poll when matchMedia is wrong, but eventually succeeds
    ● "after each" hook for "should poll when matchMedia is wrong, but eventually succeeds"


[14:49:16] HeadlessChrome 78.0.3904 (Linux 0.0.0): Executed 4 of 4 (Skipped 0) 2 FAILED


Finished in 2.169 secs / 2.154 secs @ 14:49:16 GMT-0500 (Eastern Standard Time)

SUMMARY:
● 2 tests completed
● 2 tests failed

FAILED TESTS:
  ie-media-bug
    ● should poll when matchMedia is wrong, but eventually succeeds
      HeadlessChrome 78.0.3904 (Linux 0.0.0)
    AssertionError: expected undefined to exist
        at Context.<anonymous> (usr/local/google/home/rsimha/src/amphtml/test/unit/test-ie-media-bug.js:98:5)

    ● "after each" hook for "should poll when matchMedia is wrong, but eventually succeeds"
      HeadlessChrome 78.0.3904 (Linux 0.0.0)
    ExpectationError: Unexpected call: setInterval(function () {}, 10) at http://localhost:9876/usr/local/google/home/rsimha/src/amphtml/src/service/ie-media-bug.js:42:26
    Expected setInterval(match(undefined), 10) once (never called)
    Expectation met: matchMedia((min-width: 319px) AND (max-width: 321px)) once
        at Object.fail (/tmp/node_modules/sinon/lib/sinon/mock-expectation.js:309:1)
        at Object.verify (/tmp/node_modules/sinon/lib/sinon/mock.js:114:1)
        at Context.<anonymous> (usr/local/google/home/rsimha/src/amphtml/test/unit/test-ie-media-bug.js:46:16)

[14:49:16] Stopped server at http://localhost:8081
[14:49:16] ERROR: Karma test failed with exit code 1
[14:49:16] Finished 'unit' after 32 s
rsimha@bobdylan (2019-11-14-GlobalSinon):~/src/amphtml$ 
```

**Comment by rsimha** [comment]
Found a fix. Some tests were using a global matcher in addition to the global sandbox object. There's nothing left to do.
